packages/overlay/src/overlay-stack.ts

```diff
@@ -43,7 +35,44 @@ export class OverlayStack {
     }
 
     private get topOverlay(): ActiveOverlay | undefined {
-        return this.overlays.slice(-1)[0];
+        for (
+            let index = document.body.children.length - 1;
+            index >= 0;
+            index--
+        ) {
+            const element = document.body.children[index];
+            if (element instanceof ActiveOverlay) {
+                return element;
+            }
+        }
+    }
+
+    private *overlays(): Generator<ActiveOverlay, void, undefined> {
+        for (const item of document.body.children) {
+            if (item instanceof ActiveOverlay) {
+                yield item;
+            }
+        }
```

Comment:
I'm really inclined to keep the member overlays array here, just to avoid this loop. Right now, every time you do a popup, of any kind (main case i'm thinking of is a tooltip), you end up enumerating every single child of the body tag. Now true, in a PWA style application this might be quite few, but in the general use case you could potentially have hundreds of elements to iterate and call `instanceof` on here. Since we know the list of overlays, and are managing that list ourselves, we should probably just cache it for speed here.

Reply:
Ok, I can accept that reasoning.